contrib/corrosion-cmake/CMakeLists.txt

```diff
@@ -25,44 +25,50 @@ function(set_rust_target)
         endif()
         message(STATUS "Set Rust target to ${Rust_CARGO_TARGET}")
     endif ()
-endfunction()
+
+    # FindRust.cmake
+    set(RUST_MODULE_PATH "${ClickHouse_SOURCE_DIR}/contrib/corrosion/cmake")
+    if (NOT "${RUST_MODULE_PATH}" IN_LIST "${CMAKE_MODULE_PATH}")
+        list (APPEND CMAKE_MODULE_PATH "${RUST_MODULE_PATH}")
+    endif()
+endmacro()
 
 # Keep in sync with toolchain in docker images
 # (must be called before find_package(Rust) and inclusion of Corrosion.cmake)
 set(Rust_TOOLCHAIN "nightly-2024-12-01")
-set(Rust_ERROR_MESSAGE "Cannot find ${Rust_TOOLCHAIN} Rust toolchain. You can install it with 'rustup toolchain install ${Rust_TOOLCHAIN}'")
+
+macro(find_rust)
+    set_rust_target()
+    if (NOT DEFINED Rust_FOUND)
+        find_package(Rust)
+    endif()
+    if (NOT Rust_FOUND)
+        message(${RECONFIGURE_MESSAGE_LEVEL} "Cannot find ${Rust_TOOLCHAIN} Rust toolchain. You can install it with 'rustup toolchain install ${Rust_TOOLCHAIN}'")
+    endif()
+endmacro()
 
 if (NOT ENABLE_LIBRARIES)
-  set(DEFAULT_ENABLE_RUST FALSE)
+    set(DEFAULT_ENABLE_RUST FALSE)
 elseif (NOT DEFINED ENABLE_RUST)
-  set_rust_target()
-  list (APPEND CMAKE_MODULE_PATH "${ClickHouse_SOURCE_DIR}/contrib/corrosion/cmake")
-  find_package(Rust)
-  if (NOT Rust_FOUND)
-    message(${RECONFIGURE_MESSAGE_LEVEL} "${Rust_ERROR_MESSAGE}")
-  endif()
-  set(DEFAULT_ENABLE_RUST ${Rust_FOUND})
+    find_rust()
```

Comment:
In case `ENABLE_RUST` is not defined, I'd expect the code to do its best effort to find the Rust toolchain and enable it or not depending on whether it can find it. However, we're throwing an error:

```
CMake Error at contrib/corrosion-cmake/CMakeLists.txt:46 (message):
  Cannot find nightly-2024-12-01 Rust toolchain.  You can install it with
  'rustup toolchain install nightly-2024-12-01'
Call Stack (most recent call first):
  contrib/corrosion-cmake/CMakeLists.txt:53 (find_rust)
```

Maybe we can pass a parameter to `find_rust` to fail or not depending on whether it can find Rust?

Reply:
I see, if we assume building with Rust should be the default, that's ok